lib/components/viewers/stop-viewer.js

```diff
@@ -282,23 +284,28 @@ class PatternRow extends Component {
       stopTimes,
       homeTimezone,
       stopViewerArriving,
+      stopViewerConfig,
       timeFormat
     } = this.props
     // sort stop times by next departure
-    let sortedStopTimes = null
-    if (stopTimes) {
-      sortedStopTimes = stopTimes.sort((a, b) => {
-        const aTime = a.serviceDay + a.realtimeDeparture
-        const bTime = b.serviceDay + b.realtimeDeparture
-        return aTime - bTime
-      })
-      // Cap the number of times shown for any Route at 5. TODO: make configurable
-      if (sortedStopTimes.length > 0) sortedStopTimes = sortedStopTimes.slice(0, 5)
-      // Do not show any patterns with no departures happening soon.
-      const timeIsOverThreshold = sortedStopTimes[0].realtimeDeparture - getHomeTime(homeTimezone) > ONE_HOUR_IN_SECONDS * 3
-      if (sortedStopTimes[0] && timeIsOverThreshold) {
-        return null
-      }
+    let sortedStopTimes = []
+    const hasStopTimes = stopTimes && stopTimes.length > 0
+    if (hasStopTimes) {
+      sortedStopTimes = stopTimes
+        .concat()
```

Comment:
Better, but there's more that could be done. Although this does now avoid modifying the data from the parent component, these values are still being created anew in each render in both this PatternRow component and also the StopViewer render function. Instead of recalculating and resorting on each render, we could create a selector that takes in the stop data and produces the exact display values we need. That way all these sorting, slicing and other operations would happen only once. However, that is all a premature optimization on arrays of a length of typically 5 elements, so I guess this will work for now.